src/macos/cpid_macos.c

```diff
@@ -0,0 +1,288 @@
+            EVP_MD_free(library_handle->sha256);
+        }
+
+        free(library_handle);
+    }
+}
+
+int cpid_make_uuid(_In_ cpid_handle_t library_handle, _In_ uint32_t pid, _In_ uint64_t creation_time_unix_epoch_seconds, _In_ uint32_t creation_time_micros_offset, _Out_ uuid_t uuid) {
+    if (NULL == library_handle || pid > MAX_PID || creation_time_micros_offset > MAX_MICROS_OFFSET || NULL == uuid) {
+        return RETURN_ERROR;
+    }
+
+    // set the process-specific information
+    library_handle->digest_input_content.pid = pid;
+    library_handle->digest_input_content.process_creation_time.unix_epoch_seconds = creation_time_unix_epoch_seconds;
+    library_handle->digest_input_content.process_creation_time.micros_offset = creation_time_micros_offset;
+
+    // initialize digest context for new digest calculation
+    if (OPEN_SSL_SUCCESS != EVP_DigestInit_ex2(library_handle->digest_context, library_handle->sha256, NULL)) {
+        return RETURN_ERROR;
+    }
+
+    // update digest with the input content
+    if (OPEN_SSL_SUCCESS != EVP_DigestUpdate(library_handle->digest_context, &library_handle->digest_input_content, sizeof(library_handle->digest_input_content))) {
+        return RETURN_ERROR;
+    }
+
+    // finalize the digest, coping the result to the destination buffer
+    // this intermediate buffer is needed since there is no OpenSSL option
+    // for only retrieving the first 128 bits of the digest
+    unsigned int digest_size = 0;
+    if (OPEN_SSL_SUCCESS != EVP_DigestFinal_ex(library_handle->digest_context, library_handle->digest_destination_buffer, &digest_size)) {
+        return RETURN_ERROR;
+    } else if (SHA256_BUFFER_SIZE != digest_size) {
+        return RETURN_ERROR;
+    }
+
+    // set the uuid version (UUIDv8)
+    library_handle->digest_destination_buffer[6] = (library_handle->digest_destination_buffer[6] & 0x0F) | 0x80;
+
+    // set the uuid variant
+    library_handle->digest_destination_buffer[8] = (library_handle->digest_destination_buffer[8] & 0x3F) | 0x80;
+
+    // copy the first 128 bits of the digest to the output buffer
+    memcpy(uuid, library_handle->digest_destination_buffer, sizeof(uuid_t));
+
+    return RETURN_SUCCESS;
+}
+
+int cpid_get_uuid(_In_ cpid_handle_t library_handle, _In_ uint32_t pid, _Out_ uuid_t uuid) {
+    if (NULL == library_handle || NULL == uuid) {
+        return RETURN_ERROR;
+    }
+
+    process_creation_time_t process_creation_time;
+
+    if (RETURN_SUCCESS != cpid_get_process_creation_time(pid, &process_creation_time)) {
+        return RETURN_ERROR;
+    }
+
+    if (RETURN_SUCCESS != cpid_make_uuid(library_handle, pid, process_creation_time.unix_epoch_seconds, process_creation_time.micros_offset, uuid)) {
+        return RETURN_ERROR;
+    }
+
+    return RETURN_SUCCESS;
+}
+
+int cpid_get_uuid_string(_In_ cpid_handle_t library_handle, _In_ uint32_t pid, _Out_ uuid_string_t uuid_string) {
+
+    if (NULL == library_handle || NULL == uuid_string) {
```

Comment:
Consider the following styling.

```suggestion
    if (!library_handle || !uuid_string) {
```

Reply:
I agree. A `NULL == foo` or `foo == NULL` test always makes me feel like I'm reading Java or C# 😄 

Note that this practice is throughout this code.

Reply:
will do